### PR TITLE
Avoids touching unchanged rp-application.conf on rebuilds

### PR DESCRIPTION
**Pull request by patrick-premont**
Before this change, repeated invocations of `docker:publishLocal` on an unchanged project would only reuse cached layers up to the `ADD` step in the Docker file. The `ADD` layer and subsequent ones had to be rebuilt from scratch because one jar changed. This `ADD` step creates a layer with all the jars so it is an expensive step, especially when you consider the `docker push` that occurs in `docker:publish`. The jar that changed was the main application jar, and it was rebuilt because rp-application.conf was systematically regenerated.

Now `rp-application.conf` is a managed resource that uses `Tracked.inputChanged` to avoid trigerring subsequent steps when it is unchanged.

Anecdotally, I saw this speedup on a `docker:publishLocal` in a multi-project build: it went from 216s to 68s. A factor of more than 3. On `docker:publish`, especially with slow connections, the impact could be greater.

This has only been tested on a single project, so I would like to request additional testing.

### CODE REVIEW

File: src/main/scala/com/lightbend/rp/sbtreactiveapp/SbtReactiveAppKeys.scala [hunk 1]
```diff
@@ -260,10 +260,10 @@ trait SbtReactiveAppKeys {
   val minikube = InputKey[Unit]("minikube")
 
   /**
-   * If non-empty (default: "application.conf"), all rp-tooling.conf resources in dependencies will be prepended to the
-   * unmanaged application.conf file, or one will be created if none exists. To disable this behavior, specify "".
+   * If non-empty (default: "application.conf"), all resources with the given name in dependencies will be concatenated into an
+   * managed rp-application.conf file. To disable this behavior, specify "".
    */
-  val prependRpConf = SettingKey[String]("rp-prepend-rp-application-conf")
+  val prependRpConf = SettingKey[String]("rp-concat-to-rp-application-conf")
```
Comment (eed3si9n) [inline src/main/scala/com/lightbend/rp/sbtreactiveapp/SbtReactiveAppKeys.scala]: Why was this changed to concat-to?

Comment (dwijnand) [inline src/main/scala/com/lightbend/rp/sbtreactiveapp/SbtReactiveAppKeys.scala]: Probably best to deprecate this key, and replace it with a "rp-concat-to-rp-application-conf" key.

Comment (patrick-premont) [reply src/main/scala/com/lightbend/rp/sbtreactiveapp/SbtReactiveAppKeys.scala]: Thanks. I hadn't noticed that the string parameter for a setting was actually used, unlike for tasks where it's a description and the val name is used.

Comment (dwijnand) [reply src/main/scala/com/lightbend/rp/sbtreactiveapp/SbtReactiveAppKeys.scala]: @patrick-premont Just to clarify: in sbt both `SettingKey` and `TaskKey` have a label string and a description string (more precisely they're defined in terms of an internal `AttributeKey` that takes a label and a description).

What's probably getting you confuses is that [`SettingKey.apply`](https://github.com/sbt/sbt/blob/v1.1.6/main-settings/src/main/scala/sbt/Structure.scala#L667-L673) and [`TaskKey.apply`](https://github.com/sbt/sbt/blob/v1.1.6/main-settings/src/main/scala/sbt/Structure.scala#L635-L641) take both label and description strings, but default the description to the empty string. On the other hand the [`settingKey` and `taskKey`](https://github.com/sbt/sbt/blob/v1.1.6/main/src/main/scala/sbt/BuildSyntax.scala#L15-L16) (and `inputKey`) macros instead **derive** the label from the `val` name and **require** the description.

This subtle difference has created some confusion, often with keys using their descriptions as their labels 😁.
